### PR TITLE
Piperw/olmoearth v1 1 v1 2

### PR DESCRIPTION
**Pull request by piperwolters**
### Summary

Integrate olmoearth v1.1 and v1.2 models. This is a possible refactor of how olmoearth v1.0 is implemented in main. 

Minimal testing code:
```
  import torch
  from torchgeo.models import (
      OlmoEarthBase_Weights,
      OlmoEarthLarge_Weights,
      OlmoEarthNano_Weights,
      OlmoEarthSmall_Weights, 
      OlmoEarthTiny_Weights,
      olmoearth_base,
      olmoearth_large,
      olmoearth_nano,
      olmoearth_small,
      olmoearth_tiny,
  )   
      
  for builder, enum in (
      (olmoearth_nano, OlmoEarthNano_Weights),
      (olmoearth_tiny, OlmoEarthTiny_Weights),
      (olmoearth_small, OlmoEarthSmall_Weights),
      (olmoearth_base, OlmoEarthBase_Weights),
      (olmoearth_large, OlmoEarthLarge_Weights),
  ):  
      builder()
      for weights in enum:
          a = builder(weights=weights).state_dict()
          b = builder(weights=weights).state_dict()
          assert all(torch.equal(a[k], b[k]) for k in a), weights
          print(f'{weights} {len(a)} params, weights applied')
```

Very open to other thoughts on how to do this integration.

### Rationale

OlmoEarth is great, but for users to take advantage of the latest and greatest, they need v1.1 and v1.2!

### Checklist

- [ X ] I have read the [Contributing docs](https://docs.torchgeo.org/en/stable/user/contributing.html)
- [ X ] I have read the [AI policy](https://github.com/torchgeo/governance/blob/main/AI-POLICY.md)

- [ ] 🟢 **No AI usage**: written by humans, for humans
- [ X ] 🟡 **AI-assisted**: AI helped with the coding, but I understand every line
- [ ] 🔴 **AI-generated**: AI did everything, review with caution

### CODE REVIEW

**Comment by adamjstewart** [comment]
One idea we didn't discuss on Slack:

* ResNet is currently modeled to have separate functions for different sizes and one function/enum for all versions
* OlmoEarth is currently modeled to have a single function for different sizes and separate functions/enums for each version

Do we want to try to standardize on one or the other? We can always deprecate one and switch to the other, don't have to worry too much about backwards compatibility. @isaaccorley thoughts as we move away from torchvision and towards timm?

**Comment by calebrob6** [comment]
I'd love to have this in main! @isaaccorley any thoughts on the timm thing (I haven't been following that)? My preference is to merge now and then standardize later, potentially with a new API.
